ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx

```diff
@@ -0,0 +1,225 @@
+import {ReactWrapperBase, withCurrentWorkspace} from 'app/utils';
+import {currentCohortStore, queryParamsStore} from 'app/utils/navigation';
+import {WorkspaceData} from 'app/utils/workspace-data';
+import {Cohort, SearchRequest} from 'generated/fetch';
+
+function colStyle(percentage: string) {
+  return {
+    flex: `0 0 ${percentage}%`,
+    maxWidth: `${percentage}%`,
+    minHeight: '1px',
+    padding: '0 0.5rem',
+    position: 'relative',
+    width: '100%'
+  } as React.CSSProperties;
+}
+
+interface Props {
+  setCohortChanged: (cohortChanged: boolean) => void;
+  setUpdatingCohort: (updatingCohort: boolean) => void;
+  setShowWarningModal: (showWarningModal: () => Promise<boolean>) => void;
+  workspace: WorkspaceData;
+}
+
+interface State {
+  loading: boolean;
+  overview: boolean;
+  criteria: SearchRequest;
+  updateCount: number;
+  cohort: Cohort;
+  minHeight: string;
+  modalPromise: Promise<boolean> | null;
+  modalOpen: boolean;
+  updateGroupListsCount: number;
+  cohortChanged: boolean;
+  searchContext: any;
+}
+
+export const CohortSearch = withCurrentWorkspace() (
+  class extends React.Component<Props, State> {
+    private subscription;
+    resolve: Function;
+    searchWrapper: HTMLDivElement;
+
+    constructor(props: any) {
+      super(props);
+      this.state = {
+        loading: false,
+        overview: false,
+        criteria: {includes: [], excludes: [], dataFilters: []},
+        updateCount: 0,
+        cohort: undefined,
+        minHeight: '10rem',
+        modalPromise:  null,
+        modalOpen: false,
+        updateGroupListsCount: 0,
+        cohortChanged: false,
+        searchContext: undefined
+      };
+      this.showWarningModal = this.showWarningModal.bind(this);
+    }
+
+    componentDidMount() {
+      const {workspace: {id, namespace}} = this.props;
+      this.subscription = queryParamsStore.subscribe(params => {
+        /* If a cohort id is given in the route, we initialize state with
+         * it */
+        const cohortId = params.cohortId;
+        if (cohortId) {
+          this.setState({loading: true});
+          cohortsApi().getCohort(namespace, id, cohortId).then(cohort => {
+            this.setState({cohort, loading: false});
+            currentCohortStore.next(cohort);
+            if (cohort.criteria) {
+              searchRequestStore.next(parseCohortDefinition(cohort.criteria));
+            }
+          });
+        } else {
+          this.setState({cohort: {criteria: `{'includes':[],'excludes':[],'dataFilters':[]}`, name: '', type: ''}});
+        }
+      });
+
+      this.subscription.add(searchRequestStore.subscribe(sr => {
+        const cohortChanged = !!this.state.cohort && this.state.cohort.criteria !== JSON.stringify(mapRequest(sr));
+        this.props.setCohortChanged(cohortChanged);
+        this.setState({
+          criteria: sr,
+          overview: sr.includes.length > 0 || sr.excludes.length > 0,
```

Comment:
can sr ever be null? also can we use more meaningful variable searchRequest or something?